src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -57,13 +60,35 @@ public class KestrelServerOptions
             {
                 if (value.HasValue && value.Value <= 0)
                 {
-                    throw new ArgumentOutOfRangeException("value", "Value must be null or a positive integer.");
+                    throw new ArgumentOutOfRangeException(nameof(value), "Value must be null or a positive integer.");
                 }
                 _maxRequestBufferSize = value;
             }
         }
 
         /// <summary>
+        /// Gets or sets the maximum allowed size for the HTTP request line.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to 8,192 bytes (8 KB).
+        /// </remarks>
+        public int MaxRequestLineSize
+        {
+            get
+            {
+                return _maxRequestLineSize;
+            }
+            set
+            {
+                if (value <= 0)
```

Comment:
What's the shortest practical request line? `GET / HTTP/1.1`? Any reason to allow the limit to be lower than 16?